### DISCUSSION TITLE
Abilitazione invio Tessera Sanitaria

### DISCUSSION THREAD

**Discussion by Alexxia5** [fattureincloud/api#442] [Q&A] [answered]
Ciao,
nella creazione della Fattura dovrei far in modo che si abilitata all'invio Tessera Sanitaria e il pagamento sia tracciato.
Ho impostato i valori  .tsCommunication(true) e .tsPagamentoTracciato(true) ma nella stampa del JSON non tornano valorizzati:

```
 IssuedDocument invoice = new IssuedDocument()
                .type(IssuedDocumentType.INVOICE)
                .entity(entity)
                .date(dettaglioFattura.getDataFattura())
                //  .number(2024000545) // se si omette il numero verrà incrementato automaticamente
                .numeration(numeration) // inserito solo per diversificare l'ambiente di collaudo e demo
                .subject(dettaglioFattura.getSubject()) // non visibile in fattura
                .visibleSubject(dettaglioFattura.getVisibleSubject()) // visibile in fattura
                .currency(new Currency().id("EUR"))
                .language(new Language()
                        .code("it")
                        .name("italiano")
                ).addItemsListItem(
                        new IssuedDocumentItemsListItem()
                                .productId(dettaglioFattura.getVociDaFAtturare().get(0).getId())
                                .code(dettaglioFattura.getVociDaFAtturare().get(0).getCode())
                                .name(dettaglioFattura.getVociDaFAtturare().get(0).getName())
                                .netPrice(dettaglioFattura.getImportoBigDecimal())
                                .discount(BigDecimal.valueOf(0))
                                .qty(BigDecimal.valueOf(1))
                                .vat(new VatType().id(TLMDConstants.FIC_IVA_O_ART_10_COMMA1)) 
                                .description(DOTT + " " + dettaglioFattura.getNomeMedico())
                ).addPaymentsListItem(
                        new IssuedDocumentPaymentsListItem()
                                .dueDate(dettaglioFattura.getDataPagamento()) //scadenze
                                .paidDate(dettaglioFattura.getDataPagamento())
                                .status(IssuedDocumentStatus.PAID)
                                .paymentAccount(new PaymentAccount().id(dettaglioFattura.getPaymentAccount()))

                ).extraData(
                        new IssuedDocumentExtraData()
                                .tsCommunication(true)
                                .tsPagamentoTracciato(true)
                )
                .paymentMethod(
                        new PaymentMethod().id(dettaglioFattura.getPaymentMethod()))
                .showPaymentMethod(true)  //Show the payment method details on the document
                ;
```

Questo è il log che viene stampato nel Result della chiamata:

```
extraData: class IssuedDocumentExtraData {
        showSofortButton: false
        multifattureSent: null
        tsCommunication: false
        tsFlagTipoSpesa: null
        tsPagamentoTracciato: null
        tsTipoSpesa: null
        tsOpposizione: null
        tsStatus: null
        tsFileId: null
        tsSentDate: null
        tsFullAmount: false
        importedBy: null
    }
```

Cosa sbaglio? 
Inoltre il mio intento è abilitare la fattura all'invio TS ma non di inviarla... sono dubbiosa se è corretto settare a True il campo tsCommunication. 

Grazie
Ale

**Comment by emajo** [comment, marked as answer]
Ciao, si, è corretto il campo ts_communication ma vanno settati anche gli altri campi: `ts_communication, ts_tipo_spesa, ts_flag_tipo_spesa, ts_pagamento_tracciato, ts_opposizione`.

Grazie,
Emanuele.

**Reply by Alexxia5** [reply]
Ho provato impostando i campi richiesti ma continua a non valorizzare nulla nella risposta

.extraData(
                        new IssuedDocumentExtraData()
                                .showSofortButton(false)
                                .tsCommunication(true)
                                .tsPagamentoTracciato(true)
                                .tsTipoSpesa("SR")
                                .tsFlagTipoSpesa(BigDecimal.valueOf(2))
                                .tsOpposizione(false)
                )

la risposta è sempre la stessa.
Potrebbe dipendere dal fatto che sto utilizzando un account di prova che non ha abilitato il servizio TS perchè non è stata ancora acquistata la licenza? 
In questo caso come posso testare la funzionalità?
Grazie 
Ale

**Reply by emajo** [reply]
Ciao @Alexxia5, sì, il servizio deve essere abilitato per settare quei campi, attualmente non abbiamo modo di testare questa funzionalità senza la licenza.

Grazie,
Emanuele.

**Reply by Alexxia5** [reply]
Ciao, 
quindi dovrò effettuare il test direttamente con l'account reale che ha un profilo Complete.
Per evitare troppi errori mi potete confermare che il settaggio dei campi è corretto per ottenere che la fattura sia:
- abilitata all'invio Tessera Sanitaria e il pagamento sia tracciato
- il pagamento sia tracciato
- il tipo spesa sia "SR-Spese Prestazioni Assistenza"

quindi l'extra Data lo potrò passare nel seguente modo:
.extraData(  new IssuedDocumentExtraData()
.showSofortButton(false)
.tsCommunication(true)
.tsPagamentoTracciato(true)
.tsTipoSpesa("SR")
.tsFlagTipoSpesa(BigDecimal.valueOf(2))
.tsOpposizione(false)
)
 
dalla documentazione non mi è chiaro a cosa serva il  .showSofortButton(boolean).

Vi ringrazio.
Ciao 
Ale

**Reply by emajo** [reply]
Ciao @Alexxia5 i campi sembrano settati correttamente, showSofortButton è un campo legacy che non abbiamo ancora rimosso ma comunque slegato dal sistema tessera sanitaria.

**Reply by valmoz** [reply]
Buongiorno @Alexxia5 
ha ancora bisogno di supporto?

Grazie
Mauro

**Comment by Alexxia5** [comment]
Buongiorno, per ora no. Si può chiudere il post.

Grazie.
